zomes/rea_economic_resource/lib/src/lib.rs

```diff
@@ -116,11 +127,22 @@ impl API for EconomicResourceZomePermissableDefault {
         // if the event is a transfer-like event, run the receiver's update first
         if let MaybeUndefined::Some(receiver_inventory) = &event.to_resource_inventoried_as {
             let inv_entry_hash: &EntryHash = receiver_inventory.as_ref();
-            resources_affected.push(handle_update_inventory_resource(
+            let (header_hash, resource_address, new_resource, prev_resource) = handle_update_inventory_resource(
                 &resource_entry_def_id,
                 &get_latest_header_hash(inv_entry_hash.clone())?,   // :TODO: temporal reduction here! Should error on mismatch and return latest valid ID
                 event.with_inventory_type(ResourceInventoryType::ReceivingInventory),
-            )?);
+            )?;
+            resources_affected.push((header_hash, resource_address, new_resource, prev_resource));
+            // update index if transfer or transfer_all_rights action
+            if (event.get_action() == "transfer" || event.get_action() == "transfer_all_rights") && new_resource.primary_accountable != prev_resource.primary_accountable {
```

Comment:
```suggestion
            if new_resource.primary_accountable != prev_resource.primary_accountable {
```
let's leave out the action logic here, and depend on the logic that's been specified in the Updateable trait, which will define whether the new value is different than the old